src/tray.ts

```diff
@@ -14,134 +14,116 @@ import { t } from '@/i18n';
   }
+};
 
+const setMacSpecificTraySettings = (trayInstance: Electron.Tray): void => {
+  trayInstance.setIgnoreDoubleClickEvents(true);
+};
+
+const getPixelRatio = (): number => {
+  return is.windows() ? screen.getPrimaryDisplay().scaleFactor || 1 : 1;
+};
+
+export const setTrayOnClick = (fn: TrayEvent): void => {
+  if (!tray) return;
   tray.removeAllListeners('click');
   tray.on('click', fn);
 };
 
-// Won't do anything on macOS since its disabled
-export const setTrayOnDoubleClick = (fn: TrayEvent) => {
-  if (!tray) {
-    return;
-  }
-
+export const setTrayOnDoubleClick = (fn: TrayEvent): void => {
+  if (!tray) return;
   tray.removeAllListeners('double-click');
   tray.on('double-click', fn);
 };
 
-export const setUpTray = (app: Electron.App, win: Electron.BrowserWindow) => {
+export const setUpTray = (
+  app: Electron.App,
+  win: Electron.BrowserWindow,
+): void => {
   if (!config.get('options.tray')) {
     tray = undefined;
     return;
   }
 
-  const { playPause, next, previous } = getSongControls(win);
-
-  const pixelRatio = is.windows()
-    ? screen.getPrimaryDisplay().scaleFactor || 1
-    : 1;
-  const defaultTrayIcon = nativeImage
-    .createFromPath(defaultTrayIconAsset)
-    .resize({
-      width: 16 * pixelRatio,
-      height: 16 * pixelRatio,
-    });
-  const pausedTrayIcon = nativeImage
-    .createFromPath(pausedTrayIconAsset)
-    .resize({
-      width: 16 * pixelRatio,
-      height: 16 * pixelRatio,
-    });
-
-  tray = new Tray(defaultTrayIcon);
+  const { playPause, next, previous }: AppWindowControls = getSongControls(win);
+  const pixelRatio = getPixelRatio();
+  const playTrayIcon = getTrayIcon(playTrayIconAsset, pixelRatio);
+  const pauseTrayIcon = getTrayIcon(pauseTrayIconAsset, pixelRatio);
 
-  tray.setToolTip(t('main.tray.tooltip.default'));
+  tray = new Tray(playTrayIcon);
+  setMacSpecificTraySettings(tray);
 
-  // MacOS only
-  tray.setIgnoreDoubleClickEvents(true);
-
-  tray.on('click', () => {
-    if (config.get('options.trayClickPlayPause')) {
-      playPause();
-    } else if (win.isVisible()) {
-      win.hide();
-      app.dock?.hide();
-    } else {
-      win.show();
-      app.dock?.show();
-    }
-  });
-
-  const template: MenuTemplate = [
-    {
-      label: t('main.tray.play-pause'),
-      click() {
-        playPause();
-      },
-    },
-    {
-      label: t('main.tray.next'),
-      click() {
-        next();
-      },
-    },
-    {
-      label: t('main.tray.previous'),
-      click() {
-        previous();
-      },
-    },
-    {
-      label: t('main.tray.show'),
-      click() {
-        win.show();
-        app.dock?.show();
-      },
-    },
+  tray.setToolTip(t('main.tray.tooltip.default'));
+  tray.on('click', () => handleTrayClick(app, win, playPause));
+
+  const showWindow = (): void => {
+    win.show();
+    app.dock?.show();
+  };
+
+  const trayMenuTemplate: MenuTemplate = [
+    { label: t('main.tray.play-pause'), click: playPause },
```

Comment:
lint

Reply:
I manually edited. Apparently some rules should be added to ESLint. It neither auto-lints or complains about this one.